### PR TITLE
Cover the code CourseTemplate with UnitTests

### LINKED ISSUE #28: Cover the code CourseTemplate with UnitTests

### PR DESCRIPTION
**Pull request by kmirik999**
Closes #28

### CODE REVIEW

File: CourseTemplate/CourseTemplate.UnitTests/UnitTest1.cs [hunk 1]
```diff
@@ -1,9 +1,109 @@
...
+		[Fact]
+		public async Task GetByIdAsync_WhenCourseExists_ShouldReturnViewCourseTemplateDto()
+		{
+			// Arrange
+			var existingCourse = new Data.CourseTemplate { Id = 1, CourseName = "ExistingCourse" };
+			var mockRepository = new Mock<ICourseTemplateRepository>();
+			mockRepository.Setup(repo => repo.GetByIdAsync(1))
+				.ReturnsAsync(existingCourse);
+
+			var service = new CourseTemplateService(mockRepository.Object);
```
Comment (Bardin08) [inline CourseTemplate/CourseTemplate.UnitTests/UnitTest1.cs:84]: Code duplication

File: CourseTemplate/CourseTemplate.UnitTests/UnitTest1.cs [hunk 2]
```diff
@@ -1,9 +1,109 @@
...
+		
+		[Fact]
+		public async Task GetByIdAsync_WhenCourseDoesNotExist_ShouldThrowNotFoundException()
+		{
+			// Arrange
+			var mockRepository = new Mock<ICourseTemplateRepository>();
+			mockRepository.Setup(repo => repo.GetByIdAsync(1))
+				.ReturnsAsync((Data.CourseTemplate)null);
+
+			var service = new CourseTemplateService(mockRepository.Object);
```
Comment (Bardin08) [inline CourseTemplate/CourseTemplate.UnitTests/UnitTest1.cs:103]: Code duplication

**Review by Bardin08 — CHANGES_REQUESTED** [review]

**Comment by Bardin08** [comment]
Also, please rename tests file to something more manful

**Review by YaroslavKSE — APPROVED** [review]
LGTM. 100% of coverege of Core Layer - thats perfect!

**Review by YaroslavKSE — APPROVED** [review]

**Review by roflmyrlok — APPROVED** [review]
tests are good, approve!

**Review by YaroslavKSE — APPROVED** [review]

**Review by YaroslavKSE — APPROVED** [review]

**Pull request merged**